### PR TITLE
Use OSCARAPI_BLOCK_ADMIN_API_ACCESS setting in urls, remove all is_staff exceptions which are still there

### PR DESCRIPTION
**Pull request by maerteijn**

### CODE REVIEW

File: oscarapi/views/basket.py [hunk 1]
```diff
@@ -285,15 +285,14 @@ class LineList(BasketPermissionMixin, generics.ListCreateAPIView):
...
     def get_queryset(self):
         pk = self.kwargs.get("pk")
-        if pk is not None:  # usually we need the lines of the basket
-            basket = self.check_basket_permission(self.request, basket_pk=pk)
-            prepped_basket = operations.assign_basket_strategy(basket, self.request)
-            return prepped_basket.all_lines()
-        elif self.request.user.is_staff:  # admin users can view a bit more
-            return super(LineList, self).get_queryset()
-        else:  # non admin users can view nothing at all here.
+        if pk is None:
```
[the remark this answers is not in the document]
Comment (specialunderwear) [reply oscarapi/views/basket.py]: Or is this some kind of default way to handle this?

File: oscarapi/permissions.py [hunk 2]
```diff
@@ -37,7 +40,7 @@ class APIAdminPermission(DjangoModelPermissions):
     }
 
     def has_permission(self, request, view):
-        if not request.user.is_staff:
+        if settings.OSCARAPI_BLOCK_ADMIN_API_ACCESS and not request.user.is_staff:
```
[the remark this answers is not in the document]
Comment (maerteijn) [reply oscarapi/permissions.py]: Added test

File: oscarapi/tests/unit/testbasket.py [hunk 3]
```diff
@@ -21,65 +21,33 @@ class BasketTest(APITest):
...
-        self.response.assertStatusEqual(403, "Anonymous users can not use the basket api to create baskets.")
+        self.response.assertStatusEqual(405, "Uers can not use the basket-list view to create baskets.")
 
         # authenticated
         self.login('nobody', 'nobody')
         data = {'owner': "http://testserver%s" % reverse('user-detail', args=[2])}
 
         self.response = self.client.post(url, json.dumps(data), content_type='application/json')
-        self.response.assertStatusEqual(403, "Authenticated regular users can not use the basket api to create baskets.")
+        self.response.assertStatusEqual(405, "Uers can not use the basket-list view to create baskets.")
```
[the remark this answers is not in the document]
Comment (maerteijn) [reply oscarapi/tests/unit/testbasket.py]: I changed the description to "It is not possible to use the basket-list view to create baskets."

File: oscarapi/tests/unit/testbasket.py [hunk 4]
```diff
@@ -21,65 +21,33 @@ class BasketTest(APITest):
...
         baskets = Basket.objects.all()
 
         self.assertFalse(baskets.exists(), "There should be no baskets yet.")
 
         # anonymous
         data = {}
-
         self.response = self.client.post(url, json.dumps(data), content_type='application/json')
-        self.response.assertStatusEqual(403, "Anonymous users can not use the basket api to create baskets.")
+        self.response.assertStatusEqual(405, "Uers can not use the basket-list view to create baskets.")
```
[the remark this answers is not in the document]
Comment (maerteijn) [reply oscarapi/tests/unit/testbasket.py]: I changed the description to "It is not possible to use the basket-list view to create baskets."

File: oscarapi/tests/unit/testbasket.py [hunk 5]
```diff
@@ -21,65 +21,33 @@ class BasketTest(APITest):
         'stockrecord', 'partner', 'option'
     ]
 
-    def test_basket_api_create(self):
-        "The basket api create command should work with regular cookie based login"
+    def test_basket_api_create_not_possible(self):
+        "it's not possible to create baskets with the basket-list view"
```
[the remark this answers is not in the document]
Comment (maerteijn) [reply oscarapi/tests/unit/testbasket.py]: Done

File: oscarapi/views/basic.py [hunk 6]
```diff
@@ -57,19 +57,17 @@ class CountryDetail(generics.RetrieveAPIView):
...
-    Retrieve all baskets that belong to the current user.
-    
-    While each user can view their own basket, for creating baskets
-    user must be assigned the correct (basket update) permissions.
+    Retrieve all baskets that belong to the current (authenticated) user.
     """
     serializer_class = BasketSerializer
     queryset = editable_baskets()
-    permission_classes = (DjangoModelPermissions,)
+    permission_classes = (permissions.RequestAllowsAccessTo,)
```
[the remark this answers is not in the document]
Comment (maerteijn) [reply oscarapi/views/basic.py]: This permission does not make sense AT ALL. I removed it.

File: oscarapi/views/basket.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (maerteijn) [reply oscarapi/views/basket.py]: All artifact code from the past. I cleaned it.

**Comment by mongkyo** [comment]
Hi, this merged error for me ```AttributeError: 'Settings' object has no attribute 'OSCARAPI_BLOCK_ADMIN_API_ACCESS'```

how to fix it?

**Pull request merged**